tools/clang/test/HLSL/rewriter/correct_rewrites/effects-syntax_gold.hlsl

```diff
@@ -1,24 +1,132 @@
 // Rewrite unchanged result:
-Texture2D tex : register(t1);
-Texture2D tex2 : register(t2);
-Texture2D texa[3];
-SamplerState samLinear : register(s7);
-sampler S : register(s1);
-SamplerComparisonState SC : register(s3);
-float4 main() : SV_Target {
+
+
+
+Texture2D tex : register(t1), tex2 : register(t2)
+
+
+
+
+
+
+
+< int foo=1; >
```

Comment:
I think part of the point of regenerating the code before was to be able to get rid of deprecated effects constructs like the ones in this file, such as: `<effect annotations>`, `{ state blocks }`, `technique {}`, and `pass {}`.  These are allowed on `-HV 2015`, but should not be allowed on newer language versions.

Should we have a rewriter mode that can do what we did before, reconstructing the source from AST without the parts that got ignored/eliminated?

Reply:
For this, do you think we'd need some sort of AST node that represents deprecated and ignored text so that it can be explicitly removed?